sleap/gui/commands.py

```
@@ -3470,13 +3481,203 @@ def do_action(context: CommandContext, params: dict):
+        # Update pruned labels to point to the new video
+        for labeled_frame in pruned_labels.labeled_frames:
+            labeled_frame.video = new_video
+
+        pruned_labels.videos = [new_video]
+
+        # Save the pruned labels
+        labels_filename = params["filename"].replace(".mp4", ".slp")
+        try:
+            pruned_labels.save(labels_filename)
+        except Exception as e:
+            raise RuntimeError(f"Failed to save labels to {labels_filename}: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)